### DISCUSSION TITLE
Is it possible to support "live-reload" ? (Save a file, then relaunch application debugging automatically ?)

### DISCUSSION THREAD

**Discussion by Ortega-Dan** [golang/vscode-go#1428] [Help]
Like other languages in vscode, when you save a code file while debugging, the debugging session stops, recompiles and starts again automatically. (I experience this with Angular and Java).
Is it possible ?

**Comment by hyangah** [comment]
Currently not possible - I think the extension even pops up a warning message if one tries to update code while a debugging session is active (I may be wrong). 🤔

The [vscode API](https://code.visualstudio.com/api/references/vscode-api#debug) allows to `stopDebugging` with `activeDebuggingSession`, and then call `startDebugging` with the previous `activeDebuggingSession`'s configuration, so technically I think it's possible to simulate the behavior (though, I don't expect it to be as smooth as the experience with Angular or Java for various reasons). Do you have any more information on how Angular/Java extensions implement the feature?

cc @suzmue

**Reply by Ortega-Dan** [reply]
Thanks for the response. 
It actually says nothing when code is saved. That's what I find unsafe, because we might think a change was somehow saved into the running code, and it doesn't even stop the debugging session. 

I certainly don't know now how they implement it. But I believe it is safer to at least stop debugging. 
I actually implemented a personal workaround with a Linux script to do it automatically from the terminal when a file is saved, but I believe I will take a look at it and possibly attempt at least the "stopDebugging" from vscode API.

I just have very short experience with vscode extensions development, but if the team believes that providing at least a warning, or stopping the session is usefull for the extension, that makes it more usable IMHO.

**Reply by suzmue** [reply]
Can you run the command "Go: Reset Global State" and select ignoreDebugWarningKey (if possible) and then run "Go: Reset Workspace State" and select ignoreDebugWarningKey (if possible) and then let me know if the warning message starts appearing again?

**Reply by Ortega-Dan** [reply]
Thanks.
It showed on the first attempt,\
![image](https://user-images.githubusercontent.com/24926168/115494085-8514ef80-a22a-11eb-8e40-9b8c96a6d4e3.png)
and then in the second it wasn't (wasn't possible as you suggested)\
![image](https://user-images.githubusercontent.com/24926168/115494147-a970cc00-a22a-11eb-945c-761a8e8daaae.png)
And yes it started showing the warning, which is good. thanks.
I anyway believe it anyway allowed the save, and it would be very nice if it warns and avoid the save until confirmation, or if it stops the debugging session.
